src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -763,8 +780,13 @@ fn do_mul(lhs: &Decimal, rhs: &Decimal) -> Res<Decimal> {
     let mut start_to = int_word_to + frac_word_to;
     let r_start = cmp::max(r_int_word_cnt + r_frac_word_cnt, 0) as usize;
     let left_stop = cmp::max(l_int_word_cnt + l_frac_word_cnt, 0) as usize;
+    let r_start = cmp::min(r_start, WORD_BUF_LEN as usize);
+    let left_stop = cmp::min(left_stop, WORD_BUF_LEN as usize);
```

Comment:
The four above lines are kind of hard to read. :(

Reply:
It's needed since if `r_start` or `left_stop` is bigger than WORD_BUF_LEN will cause panic. 
And I think the whole function is hard to read, you need to think  very very carefully during reading the functions in `decimal`

Reply:
Add specification comments to these functions, i.e. what it is expect to do in detail, will be more helpful..

Reply:
@AndreMouche Yes it's definitely a head scratcher. :)

Reply:
Can we use num::clamp here?